packages/jest-transform/src/shouldInstrument.ts

```diff
@@ -35,7 +35,11 @@ export default function shouldInstrument(
   if (
     !config.testPathIgnorePatterns.some(pattern => !!filename.match(pattern))
   ) {
-    if (config.testRegex.some(regex => new RegExp(regex).test(filename))) {
+    if (
+      config.testRegex.some((regex: string | RegExp) =>
```

Comment:
do we need the type here? shouldn't that be inferred by TS since we're calling this on `config.testRegex`?

Reply:
I saw ts complains about this so I had to explicitly declare type.
![image](https://user-images.githubusercontent.com/27772165/78778931-5ff56c00-799c-11ea-93a3-0a335c6b7574.png)

Reply:
hmm, odd. seems like something isnæt hooked up correctly?

Reply:
very weird

![image](https://user-images.githubusercontent.com/1404810/78779287-ffb2fa00-799c-11ea-9f20-d6274c23dce2.png)

![image](https://user-images.githubusercontent.com/1404810/78779301-06da0800-799d-11ea-87ce-c3e51d0cd6fb.png)

Reply:
It works if we do `testRegex: Array<string | RegExp>;` rather than `testRegex: Array<string> | Array<RegExp>;`

Reply:
yes like the code below. Besides shall I create a separate type declaration like 
```
// Config.ts
export type TestRegex = Array<string | RegExp>
```
and reuse it ?